### PR TITLE
fix: Fix broken proto conversion methods for data sources

### LINKED ISSUE #2581: Duplicate data source in feature registry
## Expected Behavior 
Registry protobuf should not have duplicate data source definitions.

## Current Behavior
Every time we run `feast apply`, we noticed that a data source source is appended to the registry protobuf with the exact same details.

```python
from feast.protos.feast.core.Registry_pb2 import Registry as RegistryProto

registry_proto = RegistryProto()

with open('registry.db', 'rb') as f: data = f.read()

print(registry_proto.FromString(data))
````

This is what the protobuf looks like after running `feast apply` 3 times in a row without making any changes to the definition.

```
entities {
  spec {
    name: "__dummy"
    value_type: STRING
    join_key: "__dummy_id"
    project: "myproject"
  }
  meta {
    created_timestamp {
      seconds: 1650485581
      nanos: 37674000
    }
    last_updated_timestamp {
      seconds: 1650485581
      nanos: 37723000
    }
  }
}
registry_schema_version: "1"
version_id: "743b0630-373b-405f-861d-3de738962d7c"
last_updated {
  seconds: 1650485581
  nanos: 37831000
}
feature_views {
  spec {
    name: "IRIS"
    project: "myproject"
    entities: "__dummy"
    features {
      name: "PETAL_LENGTH"
      value_type: FLOAT
    }
    features {
      name: "PETAL_WIDTH"
      value_type: FLOAT
    }
    features {
      name: "SEPAL_LENGTH"
      value_type: FLOAT
    }
    features {
      name: "SEPAL_WIDTH"
      value_type: FLOAT
    }
    features {
      name: "SPECIES"
      value_type: INT64
    }
    ttl {
      seconds: 31449600
    }
    batch_source {
      type: BATCH_SNOWFLAKE
      timestamp_field: "EVENT_TIMESTAMP"
      created_timestamp_column: "CREATE_TIMESTAMP"
      data_source_class_type: "feast.infra.offline_stores.snowflake_source.SnowflakeSource"
      snowflake_options {
        table: "IRIS"
        schema: "MY_SCHEMA"
        database: "MY_DATABASE"
      }
    }
    online: true
    description: "A sample feature view containing the Iris dataset."
  }
  meta {
    created_timestamp {
      seconds: 1650485581
      nanos: 37333000
    }
    last_updated_timestamp {
      seconds: 1650485581
      nanos: 37333000
    }
  }
}
data_sources {
  type: BATCH_SNOWFLAKE
  timestamp_field: "EVENT_TIMESTAMP"
  created_timestamp_column: "CREATE_TIMESTAMP"
  data_source_class_type: "feast.infra.offline_stores.snowflake_source.SnowflakeSource"
  snowflake_options {
    table: "IRIS"
    schema: "MY_SCHEMA"
    database: "MY_DATABASE"
  }
  project: "myproject"
}
data_sources {
  type: BATCH_SNOWFLAKE
  timestamp_field: "EVENT_TIMESTAMP"
  created_timestamp_column: "CREATE_TIMESTAMP"
  data_source_class_type: "feast.infra.offline_stores.snowflake_source.SnowflakeSource"
  snowflake_options {
    table: "IRIS"
    schema: "MY_SCHEMA"
    database: "MY_DATABASE"
  }
  project: "myproject"
}
data_sources {
  type: BATCH_SNOWFLAKE
  timestamp_field: "EVENT_TIMESTAMP"
  created_timestamp_column: "CREATE_TIMESTAMP"
  data_source_class_type: "feast.infra.offline_stores.snowflake_source.SnowflakeSource"
  snowflake_options {
    table: "IRIS"
    schema: "MY_SCHEMA"
    database: "MY_DATABASE"
  }
  project: "myproject"
}
```
## Steps to reproduce

Here's the `feature_store.yaml`:
```yaml
project: myproject
registry: s3://mybucket/registry.db
provider: local
offline_store:
  type: snowflake.offline
  account: myaccount
  user: myuser
  password: mypasssword
  role: myrole
  warehouse: mywarehouse
  database: mydatabase
online_store:
  type: redis
  connection_string: myredisconnectionstring
```

And here's the feature definition `feature_definitions/iris.py`:
```python
from datetime import timedelta
from pathlib import Path

from feast import Feature, FeatureView, SnowflakeSource, ValueType

name = Path(__file__).stem

iris_source = SnowflakeSource(
    schema="MY_SCHEMA",
    database="MY_DATABASE",
    table="IRIS",
    timestamp_field="EVENT_TIMESTAMP",
    created_timestamp_column="CREATE_TIMESTAMP"
)

iris = FeatureView(
    name=name.upper(),
    entities=[],
    ttl=timedelta(weeks=52),
    schema=[
        Feature(name="PETAL_LENGTHS", dtype=ValueType.FLOAT),
        Feature(name="PETAL_WIDTH", dtype=ValueType.FLOAT),
        Feature(name="SEPAL_LENGTH", dtype=ValueType.FLOAT),
        Feature(name="SEPAL_WIDTH", dtype=ValueType.FLOAT),
        Feature(name="SPECIES", dtype=ValueType.INT64),
    ],
    online=True,
    source=iris_source,
    description="A sample feature view containing the Iris dataset."
)
```

### Specifications

- Version: `0.2.0`
- Platform: AWS EC2 (Linux2 AMI)
- Subsystem:

### PR DESCRIPTION
**Pull request by felixwang9817**
**What this PR does / why we need it**: The Redshift and Snowflake data sources did not implement conversion to and from protos correctly. This PR fixes that logic and adds tests. It also cleans up a bunch of data source classes by removing unnecessary properties.

**Which issue(s) this PR fixes**:

Fixes #2581

### CODE REVIEW

File: sdk/python/feast/infra/offline_stores/snowflake_source.py [hunk 1]
```diff
@@ -183,6 +179,7 @@ def to_proto(self) -> DataSourceProto:
             A DataSourceProto object.
         """
         data_source_proto = DataSourceProto(
+            name=self.name,
```
Comment (achals) [inline sdk/python/feast/infra/offline_stores/snowflake_source.py:182]: Was this the main bug leading to duplicate data sources?

File: sdk/python/tests/unit/test_data_sources.py [hunk 2]
```diff
@@ -145,3 +149,107 @@ def test_default_data_source_kw_arg_warning():
...
+    )
+
+    assert DataSource.from_proto(bigquery_source.to_proto()) == bigquery_source
+    assert DataSource.from_proto(file_source.to_proto()) == file_source
+    assert DataSource.from_proto(redshift_source.to_proto()) == redshift_source
+    assert DataSource.from_proto(snowflake_source.to_proto()) == snowflake_source
+    assert DataSource.from_proto(kafka_source.to_proto()) == kafka_source
+    assert DataSource.from_proto(kinesis_source.to_proto()) == kinesis_source
+    assert DataSource.from_proto(push_source.to_proto()) == push_source
+    assert DataSource.from_proto(request_source.to_proto()) == request_source
```
Comment (achals) [inline sdk/python/tests/unit/test_data_sources.py:255]: Nice

**Review by achals — APPROVED** [review]
/lgtm

File: sdk/python/feast/infra/offline_stores/snowflake_source.py (hunk 1 above)
Comment (felixwang9817) [reply sdk/python/feast/infra/offline_stores/snowflake_source.py:182]: yes this was the root cause, see https://github.com/feast-dev/feast/issues/2581#issuecomment-1107694083

Comment (achals) [reply sdk/python/feast/infra/offline_stores/snowflake_source.py:182]: Good catch.

**Comment by achals** [comment]
We should patch release this soon-ish.

**Pull request merged**